client/src/main/java/io/pravega/client/connection/impl/RawClient.java

```diff
@@ -87,16 +88,24 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         }
     }
 
-    public RawClient(PravegaNodeUri uri, ConnectionPool connectonPool) {
+    public RawClient(PravegaNodeUri uri, ConnectionPool connectionPool) {
         this.segmentId = null;
-        this.connection = connectonPool.getClientConnection(flow, uri, responseProcessor);
+        this.connection = connectionPool.getClientConnection(flow, uri, responseProcessor)
+        .exceptionally(e -> {
+            log.warn("Exception observed while attempting to obtain a connection to segment store {}", uri, e);
+            throw new CompletionException(new ConnectionFailedException("Failed to obtain client connection to  " + uri));
```

Comment:
I would add 'e' as cause

Reply:
done. ( i was about to push that change :D )